pkg/registry/interface.go

```diff
@@ -0,0 +1,23 @@
+package registry
+
+import (
+	"context"
+
+	"github.com/notaryproject/notation-go"
+	"github.com/opencontainers/go-digest"
+)
+
+// SignatureRepository provides a storage for signatures
+type SignatureRepository interface {
+	// Lookup finds all signatures for the specified manifest
+	Lookup(ctx context.Context, manifestDigest digest.Digest) ([]digest.Digest, error)
+
+	// Get downloads the signature by the specified digest
+	Get(ctx context.Context, signatureDigest digest.Digest) ([]byte, error)
+
+	// Put uploads the signature to the registry
+	Put(ctx context.Context, signature []byte) (notation.Descriptor, error)
```

Comment:
The suffix `SignatureBlob` is redundant as the parameter and documentation show it's a signature blob.

Reply:
What you said makes sense if the `Put` function is on `SignatureBlob` struct. Since this is an interface for `SignatureRepository` that provides methods for manifests and signatures, we can use bit descriptive names.